tutorials/models_mapping/plot_1_tensor_models.py

```diff
@@ -0,0 +1,337 @@
+
+
+"""
+Tensor Meshes
+=============
+
+Here we demonstrate various ways that models can be defined and mapped to
+tensor meshes. Some things we consider are:
+
+    - Surface topography
+    - Adding structures of various shape to the model
+    - Parameterized models
+    - Models with 2 or more physical properties
+    
+
+"""
+
+#########################################################################
+# Import modules
+# --------------
+#
+
+from discretize import TensorMesh
+from SimPEG.utils import mkvc, surface2ind_topo, ModelBuilder
+from SimPEG import maps
+import numpy as np
+import matplotlib.pyplot as plt
+
+# sphinx_gallery_thumbnail_number = 3
+
+#############################################
+# Defining the mesh
+# -----------------
+#
+# Here, we create the tensor mesh that will be used for all examples.
+#
+
+
+def make_example_mesh():
+
+    dh = 5.
+    hx = [(dh, 5, -1.3), (dh, 20), (dh, 5, 1.3)]
+    hy = [(dh, 5, -1.3), (dh, 20), (dh, 5, 1.3)]
+    hz = [(dh, 5, -1.3), (dh, 20), (dh, 5, 1.3)]
+    mesh = TensorMesh([hx, hy, hz], 'CCC')
+
+    return mesh
+
+
+#############################################
+# Halfspace model with topography at z = 0
+# ----------------------------------------
+#
+# In this example we generate a half-space model. Since air cells remain
+# constant during geophysical inversion, the number of model values we define
+# should be equal to the number of cells lying below the surface. Here, we
+# define the model (*mod* ) as well as the mapping (*mod_map* ) that goes from
```

Comment:
Remove this sentence and use plain English for variable names, even if it's long.

Reply:
I changed the variable names, but I like giving context to what is happening.